### PR TITLE
Order

### PR DESCRIPTION
**Pull request by DevIgork**

### CODE REVIEW

**Review by Alexander1iubchenko — APPROVED** [review]
Looks good to me. Good job

File: src/main/java/com/example/bookstore/model/OrderItem.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+@Where(clause = "is_deleted=false")
+@Table(name = "order_items")
+public class OrderItem {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne(fetch = FetchType.LAZY)
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    @JoinColumn(name = "order_id",nullable = false)
```
Comment (yevheniisynytsia) [inline src/main/java/com/example/bookstore/model/OrderItem.java]:
```suggestion
    @JoinColumn(name = "order_id", nullable = false)
```

File: src/main/java/com/example/bookstore/model/OrderItem.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    private Order order;
+    @ManyToOne(fetch = FetchType.LAZY)
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    @JoinColumn(name = "book_id",nullable = false)
```
Comment (yevheniisynytsia) [inline src/main/java/com/example/bookstore/model/OrderItem.java]:
```suggestion
    @JoinColumn(name = "book_id", nullable = false)
```

File: src/main/resources/db/changelog/changes/18-create-order-items-table.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+databaseChangeLog:
+  - changeSet:
+      id: create-orders-table
```
Comment (yevheniisynytsia) [inline src/main/resources/db/changelog/changes/18-create-order-items-table.yaml]:
```suggestion
      id: create-order-items-table
```

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 4]
```diff
@@ -31,3 +31,9 @@ databaseChangeLog:
       file: db/changelog/changes/15-create-shopping-cart-table.yaml
   - include:
       file: db/changelog/changes/16-create-cart-items-table.yaml
+  - include:
+      file: db/changelog/changes/17-create-orders-table.yaml
+  - include:
+      file: db/changelog/changes/18-create-order-items-table.yaml
+  - include:
+      file: db/changelog/changes/19-insert-user-roles-table.yaml
```
Comment (yevheniisynytsia) [inline src/main/resources/db/changelog/db.changelog-master.yaml]: add empty line

**Review by yevheniisynytsia — CHANGES_REQUESTED** [review]
Well done! Minor style comments left

File: src/main/java/com/example/bookstore/controller/OrderController.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        User user = (User) authentication.getPrincipal();
+        return orderService.getAll(user.getId());
+    }
+
+    @Operation(summary = "Patch status", description = "change status of order")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PatchMapping("{id}")
+    public OrderDto updateStatus(
+            @PathVariable Long id,
+            @RequestBody UpdateOrderStatusDto updateOrderStatusDto
```
Comment (d1sam) [inline src/main/java/com/example/bookstore/controller/OrderController.java]:
```suggestion
            @RequestBody @Valid UpdateOrderStatusDto updateOrderStatusDto
```

File: src/main/java/com/example/bookstore/service/impl/OrderServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+                .orElseThrow(() ->
+                        new EntityNotFoundException("Can't find user cart by id"
+                        + userId));
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find user by id" + userId));
+        String shippingAddress = requestDto.getShippingAddress();
+        Order order = new Order();
+        order.setOrderDate(LocalDateTime.now());
+        order.setUser(user);
+        order.setStatus(Order.Status.valueOf(PENDING.name()));
```
Comment (d1sam) [inline src/main/java/com/example/bookstore/service/impl/OrderServiceImpl.java]:
```suggestion
        order.setStatus(PENDING);
```

File: src/main/java/com/example/bookstore/service/impl/OrderServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    private final CartItemRepository cartItemRepository;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final UserRepository userRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderMapper orderMapper;
+
+    @Transactional
+    @Override
+    public OrderDto create(AddOrderRequestDto requestDto, Long userId) {
```
Comment (d1sam) [inline src/main/java/com/example/bookstore/service/impl/OrderServiceImpl.java:42]: try to divide logic in this method to smaller private methods

File: src/main/java/com/example/bookstore/service/impl/OrderServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+        orderRepository.save(order);
+        orderItemRepository.saveAll(orderItems);
+        return orderMapper.toDto(order);
+    }
+
+    @Override
+    public List<OrderDto> getAll(Long userId) {
+        List<Order> orders = orderRepository.findAllByUserId(userId);
+        return orders.stream()
+                .map(orderMapper::toDto)
```
Comment (d1sam) [inline src/main/java/com/example/bookstore/service/impl/OrderServiceImpl.java]: it's better to use method annotated as `@IterableMapping` inside mapper for mappings of `Iterable` objects

**Review by d1sam — CHANGES_REQUESTED** [review]
Good job, but there are some minor changes should be done!

**Review by yevheniisynytsia — APPROVED** [review]
well done!

**Pull request merged**